pandas/io/excel/_odswriter.py

```diff
@@ -0,0 +1,217 @@
+from collections import defaultdict
+import datetime
+from typing import DefaultDict, Dict
+
+import pandas._libs.json as json
+
+from pandas.io.excel._base import ExcelWriter
+from pandas.io.excel._util import _validate_freeze_panes
+
+
+class _ODSWriter(ExcelWriter):
+    engine = "odf"
+    supported_extensions = (".ods",)
+
+    def __init__(self, path, engine=None, encoding=None, mode="w", **engine_kwargs):
+        from odf.opendocument import OpenDocumentSpreadsheet
+
+        engine_kwargs["engine"] = engine
+
+        if mode == "a":
+            raise ValueError("Append mode is not supported with odf!")
+
+        super().__init__(path, mode=mode, **engine_kwargs)
+
+        self.book = OpenDocumentSpreadsheet()
+        self._style_dict: Dict[str, str] = {}
+
+    def save(self):
```

Comment:
can you type as much as possible (e.g. -> None)

Reply:
added type info